gtsam/linear/NoiseModel.cpp

```diff
@@ -238,6 +238,25 @@ void Gaussian::WhitenSystem(Matrix& A1, Matrix& A2, Matrix& A3, Vector& b) const
 
 Matrix Gaussian::information() const { return R().transpose() * R(); }
 
+/* *******************************************************************************/
+double Gaussian::logNormalizationConstant() const {
+  // Since noise models are Gaussian, we can get the logDeterminant easily
+  // Sigma = (R'R)^{-1}, det(Sigma) = det((R'R)^{-1}) = det(R'R)^{-1}
+  // log det(Sigma) = -log(det(R'R)) = -2*log(det(R))
+  // Hence, log det(Sigma)) = -2.0 * logDeterminant()
+  // which gives log = -0.5*n*log(2*pi) - 0.5*(-2.0 * logDeterminant())
+  //     = -0.5*n*log(2*pi) + (0.5*2.0 * logDeterminant())
+  //     = -0.5*n*log(2*pi) + logDeterminant()
+  double logDetR =
+      R().diagonal().unaryExpr([](double x) { return log(x); }).sum();
```

Comment:
This is very expensive in case of diagonal, isotropic, unit, because R() creates a *full* matrix. You should override in Diagonal and use invsigmas() there. In Isotropic, there is an even simpler expression, and in Unit even simpler.

Reply:
I was going to do this in a follow up PR so less to review, but I'll put it here.

Reply:
Done. I defined a new protected class method `logDetR` which isolates and takes advantage of computational efficiencies. This way we can only define `logNormalizationConstant` once and avoid duplication.